### PR TITLE
Document KEB in Compass

### PR DESCRIPTION
**Pull request by klaudiagrz**
**Description**

Changes proposed in this pull request:

- Add KEB overview
- Add KEB architecture together with the diagram
- Update README.md

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/README.md [hunk 1]
```diff
@@ -2,21 +2,33 @@
...
-| **APP_PROVISIONING_SECRET_NAME** | No | | Specifies the name of the Secret which holds credentials to the Provisioner API. |
-| **APP_PROVISIONING_GARDENER_PROJECT_NAME** | No | `true` | Defines the used Gardener project name. |
-| **APP_PROVISIONING_GCP_SECRET_NAME** | No | | Defines the name of the Secret which holds credentials to GCP. |
-| **APP_PROVISIONING_AWS_SECRET_NAME** | No | | Defines the name of the Secret which holds credentials to AWS. |
-| **APP_PROVISIONING_AZURE_SECRET_NAME** | No | | Defines the name of the Secret which holds credentials to Azure. |
-| **APP_AUTH_USERNAME** | No | | Specifies the Kyma Environment Service Broker authentication username. |
-| **APP_AUTH_PASSWORD** | No | | Specifies the Kyma Environment Service Broker authentication password. |
+| Name | Description | Default value |
+|-----|---------|:--------:|
+| **APP_PORT** | The port on which the HTTP server listens. | `8080` |
```
Comment (kazydek) [inline components/kyma-environment-broker/README.md]: I would be consistent in descriptions and start all wither with a noun or with a verb

File: components/kyma-environment-broker/README.md [hunk 2]
```diff
@@ -2,21 +2,33 @@
...
+| **APP_PROVISIONING_URL** | Specifies an URL to the provisioner API. | None |
+| **APP_PROVISIONING_SECRET_NAME** | Specifies the name of the Secret which holds credentials to the Provisioner API. | None |
+| **APP_PROVISIONING_GARDENER_PROJECT_NAME** | Defines the used Gardener project name. | `true` |
```
Comment (kazydek) [inline components/kyma-environment-broker/README.md]:
```suggestion
| **APP_PROVISIONING_GARDENER_PROJECT_NAME** | Defines the Gardener project name. | `true` |
```

Comment (kazydek) [reply components/kyma-environment-broker/README.md]: or `used`/`in use` at the end?

File: components/kyma-environment-broker/README.md [hunk 3]
```diff
@@ -2,21 +2,33 @@
...
+| **APP_PROVISIONING_GCP_SECRET_NAME** | Defines the name of the Secret which holds credentials to GCP. | None |
+| **APP_PROVISIONING_AWS_SECRET_NAME** | Defines the name of the Secret which holds credentials to AWS. | None |
+| **APP_PROVISIONING_AZURE_SECRET_NAME** | Defines the name of the Secret which holds credentials to Azure. | None |
+| **APP_AUTH_USERNAME** | Specifies the Kyma Environment Service Broker authentication username. | None |
+| **APP_AUTH_PASSWORD** | Specifies the Kyma Environment Service Broker authentication password. | None |
+| **APP_DIRECTOR_NAMESPACE** | Specifies the Namespace in which Director is deployed. | `compass-system` |
+| **APP_DIRECTOR_URL** | Specifies Director's URL. | `http://compass-director.compass-system.svc.cluster.local:3000/graphql |
```
Comment (kazydek) [inline components/kyma-environment-broker/README.md]:
```suggestion
| **APP_DIRECTOR_URL** | Specifies the Director's URL. | `http://compass-director.compass-system.svc.cluster.local:3000/graphql |
```

File: components/kyma-environment-broker/README.md [hunk 4]
```diff
@@ -2,21 +2,33 @@
...
+| **APP_DIRECTOR_OAUTH_CREDENTIALS_SECRET_NAME** | Specifies the name of the Secret created by the Integration System. | `compass-kyma-environment-broker-credentials` |
+| **APP_DIRECTOR_SKIP_CERT_VERIFICATION** | Specifies whether TLS checks the presented certificates. | `false` |
+| **APP_DATABASE_USER** | Database username | `postgres` |
```
Comment (kazydek) [inline components/kyma-environment-broker/README.md]: I guess those should also start with a verb

File: components/kyma-environment-broker/README.md [hunk 5]
```diff
@@ -2,21 +2,33 @@
...
+| **APP_DATABASE_PASSWORD** | Database user password | `password` |
+| **APP_DATABASE_HOST** | Database host | `localhost` |
+| **APP_DATABASE_PORT** | Database port | `5432` |
+| **APP_DATABASE_NAME** | Database name | `broker` |
+| **APP_DATABASE_SSL** | SSL Mode for PostgrSQL. See all the possible values [here](https://www.postgresql.org/docs/9.1/libpq-ssl.html)  | `disable`|
```
Comment (kazydek) [inline components/kyma-environment-broker/README.md]:
```suggestion
| **APP_DATABASE_SSL** | SSL Mode for PostgrSQL. See all the possible values [here](https://www.postgresql.org/docs/9.1/libpq-ssl.html).  | `disable`|
```

File: docs/kyma-environment-broker/02-01-architecture.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+The diagram and steps describe the Kyma Environment Broker (KEB) workflow and the roles of specific components in this process:
+
+![KEB diagram](./assets/keb-architecture.svg)
+
+1. The user sends a request to create a new cluster with [Kyma Runtime](https://github.com/kyma-incubator/compass/blob/master/docs/compass/02-01-components.md#kyma-runtime).
+2. KEB proxies the request to create a new cluster to the Provisioner component.
+3. Provisioner registers a new cluster in the Director component.
+4. Provisioner creates a new cluster and installs Kyma Runtime.
+5. When the operation is successful, KEB keeps sending a request for a Dashboard URL to Director:
```
Comment (kazydek) [inline docs/kyma-environment-broker/02-01-architecture.md]: There needs to be an empty line as points won't render well

File: docs/kyma-environment-broker/02-01-architecture.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+    a. KEB sends a request to Hydra to refresh the OAuth token.
+    b. KEB passes the OAuth token through Gateway to Director.
+    c. Director returns the Dashboard URL, which is the URL to the newly created cluster, through Gateway to KEB.
```
Comment (kazydek) [inline docs/kyma-environment-broker/02-01-architecture.md]:
```suggestion
    c. Director returns the Dashboard URL through Gateway to KEB. The Dashboard URL is the URL to the newly created cluster.
```

File: components/kyma-environment-broker/README.md [hunk 8]
```diff
@@ -2,21 +2,33 @@
 
 ## Overview
 
-Kyma Environment Broker provides a way to run Kyma as a Runtime on clusters provided by different cloud providers, such as AWS, GCP or Azure. The broker provides services that install Kyma with a separate plan for each provider.
-It uses the Compass Provisioner API to install Kyma on the given cluster.
+Kyma Environment Broker is a component that allows you to run Kyma as a Runtime on clusters provided by third-party providers. It uses Provisioner's API to install Kyma on a given cluster.
```
Comment (polskikiel) [inline components/kyma-environment-broker/README.md]: it allows me to **provision** Kyma as a Runtime

File: docs/kyma-environment-broker/02-01-architecture.md [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+![KEB diagram](./assets/keb-architecture.svg)
+
+1. The user sends a request to create a new cluster with [Kyma Runtime](https://github.com/kyma-incubator/compass/blob/master/docs/compass/02-01-components.md#kyma-runtime).
+2. KEB proxies the request to create a new cluster to the Provisioner component.
+3. Provisioner registers a new cluster in the Director component.
+4. Provisioner creates a new cluster and installs Kyma Runtime.
+5. When the operation is successful, KEB keeps sending a request for a Dashboard URL to Director:
+
+    a. KEB sends a request to Hydra to refresh the OAuth token.
```
Comment (polskikiel) [inline docs/kyma-environment-broker/02-01-architecture.md]: only if it's needed

Comment (klaudiagrz) [inline docs/kyma-environment-broker/02-01-architecture.md]:
```suggestion
    a. KEB sends a request to Hydra to refresh the OAuth token, if needed.
```

File: components/kyma-environment-broker/README.md (hunk 8 above)
Comment (klaudiagrz) [inline components/kyma-environment-broker/README.md]:
```suggestion
Kyma Environment Broker is a component that allows you to provision Kyma as a Runtime on clusters provided by third-party providers. It uses Provisioner's API to install Kyma on a given cluster.
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+These are the provisioning parameters for this plan:
+
+| Parameter Name | Type | Description | Required | Default value |
+|----------------|-------|-------------|:----------:|---------------|
+| **name** | string | Specifies the name of the cluster. | Yes | None |
+| **nodeCount** | int | Specifies the number of Nodes in a cluster. | No | `3` |
+| **volumeSizeGb** | int | Specifies the size of the root volume. | No | `50` |
+| **machineType** | string | Specifies the provider-specific Virtual Machine type. | No | `Standard_D2_v3` |
+| **region** | string | Defines the cluster region. The possible values are `westeurope`, `eastus`, `eastus2`, `centralus`, `northeurope`, `southeastasia`, `japaneast`, `westus2`, and `uksouth`. | No | `westeurope` |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md]:
```suggestion
| **region** | string | Defines the cluster region. | No | `westeurope` |
```

File: docs/kyma-environment-broker/01-01-overview.md [hunk 11]
```diff
@@ -0,0 +1,3 @@
+# Kyma Environment Broker
+
+Kyma Environment Broker is a component that allows you to run Kyma as a Runtime on clusters provided by third-party providers. It uses Provisioner's API to install Kyma on a given cluster.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/01-01-overview.md]:
```suggestion
Kyma Environment Broker is a component that allows you to provision Kyma as a Runtime on clusters provided by third-party providers. It uses Provisioner's API to install Kyma on a given cluster. 
```

File: components/kyma-environment-broker/README.md [hunk 12]
```diff
@@ -2,21 +2,33 @@
...
-| **APP_PROVISIONING_GCP_SECRET_NAME** | No | | Defines the name of the Secret which holds credentials to GCP. |
-| **APP_PROVISIONING_AWS_SECRET_NAME** | No | | Defines the name of the Secret which holds credentials to AWS. |
-| **APP_PROVISIONING_AZURE_SECRET_NAME** | No | | Defines the name of the Secret which holds credentials to Azure. |
-| **APP_AUTH_USERNAME** | No | | Specifies the Kyma Environment Service Broker authentication username. |
-| **APP_AUTH_PASSWORD** | No | | Specifies the Kyma Environment Service Broker authentication password. |
+| Name | Description | Default value |
+|-----|---------|:--------:|
+| **APP_PORT** | Specifies the port on which the HTTP server listens. | `8080` |
+| **APP_PROVISIONING_URL** | Specifies a URL to the Provisioner's API. | None |
+| **APP_PROVISIONING_SECRET_NAME** | Specifies the name of the Secret which holds credentials to the Provisioner API. | None |
```
Comment (kazydek) [inline components/kyma-environment-broker/README.md]: `Provisioner's API` as above?

File: components/kyma-environment-broker/README.md [hunk 13]
```diff
@@ -2,21 +2,33 @@
...
+| **APP_PROVISIONING_GCP_SECRET_NAME** | Defines the name of the Secret which holds credentials to GCP. | None |
+| **APP_PROVISIONING_AWS_SECRET_NAME** | Defines the name of the Secret which holds credentials to AWS. | None |
+| **APP_PROVISIONING_AZURE_SECRET_NAME** | Defines the name of the Secret which holds credentials to Azure. | None |
+| **APP_AUTH_USERNAME** | Specifies the Kyma Environment Service Broker authentication username. | None |
+| **APP_AUTH_PASSWORD** | Specifies the Kyma Environment Service Broker authentication password. | None |
+| **APP_DIRECTOR_NAMESPACE** | Specifies the Namespace in which Director is deployed. | `compass-system` |
+| **APP_DIRECTOR_URL** | Specifies the Director's URL. | `http://compass-director.compass-system.svc.cluster.local:3000/graphql` |
+| **APP_DIRECTOR_OAUTH_CREDENTIALS_SECRET_NAME** | Specifies the name of the Secret created by the Integration System. | `compass-kyma-environment-broker-credentials` |
+| **APP_DIRECTOR_SKIP_CERT_VERIFICATION** | Specifies whether TLS checks the presented certificates. | `false` |
+| **APP_DATABASE_USER** | Defines database username. | `postgres` |
```
Comment (kazydek) [inline components/kyma-environment-broker/README.md]:
```suggestion
| **APP_DATABASE_USER** | Defines the database username. | `postgres` |
```

File: components/kyma-environment-broker/README.md [hunk 14]
```diff
@@ -2,21 +2,33 @@
...
+| **APP_DATABASE_PASSWORD** | Defines database user password | `password` |
```
Comment (kazydek) [inline components/kyma-environment-broker/README.md]:
```suggestion
| **APP_DATABASE_PASSWORD** | Defines the database user password | `password` |
```

File: components/kyma-environment-broker/README.md [hunk 15]
```diff
@@ -2,21 +2,33 @@
...
+| **APP_DATABASE_HOST** | Defines database host | `localhost` |
```
Comment (kazydek) [inline components/kyma-environment-broker/README.md]:
```suggestion
| **APP_DATABASE_HOST** | Defines the database host | `localhost` |
```

File: components/kyma-environment-broker/README.md [hunk 16]
```diff
@@ -2,21 +2,33 @@
...
+| **APP_DATABASE_PORT** | Defines database port | `5432` |
```
Comment (kazydek) [inline components/kyma-environment-broker/README.md]:
```suggestion
| **APP_DATABASE_PORT** | Defines the database port | `5432` |
```

File: components/kyma-environment-broker/README.md [hunk 17]
```diff
@@ -2,21 +2,33 @@
...
+| **APP_DATABASE_NAME** | Defines database name | `broker` |
```
Comment (kazydek) [inline components/kyma-environment-broker/README.md]:
```suggestion
| **APP_DATABASE_NAME** | Defines the database name | `broker` |
```

File: components/kyma-environment-broker/README.md [hunk 18]
```diff
@@ -2,21 +2,33 @@
...
+| **APP_DATABASE_SSL** | Specifies SSL Mode for PostgrSQL. See all the possible values [here](https://www.postgresql.org/docs/9.1/libpq-ssl.html).  | `disable`|
```
Comment (kazydek) [inline components/kyma-environment-broker/README.md]:
```suggestion
| **APP_DATABASE_SSL** | Specifies the SSL Mode for PostgrSQL. See all the possible values [here](https://www.postgresql.org/docs/9.1/libpq-ssl.html).  | `disable`|
```

File: docs/kyma-environment-broker/02-01-architecture.md [hunk 19]
```diff
@@ -0,0 +1,17 @@
...
+
+1. The user sends a request to create a new cluster with [Kyma Runtime](https://github.com/kyma-incubator/compass/blob/master/docs/compass/02-01-components.md#kyma-runtime).
+2. KEB proxies the request to create a new cluster to the Provisioner component.
+3. Provisioner registers a new cluster in the Director component.
+4. Provisioner creates a new cluster and installs Kyma Runtime.
+5. When the operation is successful, KEB keeps sending a request for a Dashboard URL to Director:
+
+    a. KEB sends a request to Hydra to refresh the OAuth token, if needed.
+
+    b. KEB passes the OAuth token through Gateway to Director.
```
Comment (kazydek) [inline docs/kyma-environment-broker/02-01-architecture.md]:
```suggestion
    b. KEB passes the OAuth token to Director through Gateway.
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 20]
```diff
@@ -0,0 +1,30 @@
...
+## Provisioning parameters
+
+These are the provisioning parameters for this plan:
+
+| Parameter Name | Type | Description | Required | Default value |
+|----------------|-------|-------------|:----------:|---------------|
+| **name** | string | Specifies the name of the cluster. | Yes | None |
+| **nodeCount** | int | Specifies the number of Nodes in a cluster. | No | `3` |
+| **volumeSizeGb** | int | Specifies the size of the root volume. | No | `50` |
+| **machineType** | string | Specifies the provider-specific Virtual Machine type. | No | `Standard_D2_v3` |
```
Comment (kazydek) [inline docs/kyma-environment-broker/03-01-service-description.md]:
```suggestion
| **machineType** | string | Specifies the provider-specific virtual machine type. | No | `Standard_D2_v3` |
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 21]
```diff
@@ -0,0 +1,30 @@
...
+| **region** | string | Defines the cluster region. | No | `westeurope` |
+| **zone** | string | Defines the cluster zone. | No | None |
+| **autoScalerMin** | int | Specifies the minimum number of Virtual Machines to create. | No | `2` |
```
Comment (kazydek) [inline docs/kyma-environment-broker/03-01-service-description.md]:
```suggestion
| **autoScalerMin** | int | Specifies the minimum number of virtual machines to create. | No | `2` |
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+| **autoScalerMax** | int | Specifies the maximum number of Virtual Machines to create. | No | `4` |
```
Comment (kazydek) [inline docs/kyma-environment-broker/03-01-service-description.md]:
```suggestion
| **autoScalerMax** | int | Specifies the maximum number of virtual machines to create. | No | `4` |
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+| **maxSurge** | int | Specifies the maximum number of Virtual Machines that are created during an update. | No | `4` |
```
Comment (kazydek) [inline docs/kyma-environment-broker/03-01-service-description.md]:
```suggestion
| **maxSurge** | int | Specifies the maximum number of virtual machines that are created during an update. | No | `4` |
```

File: components/kyma-environment-broker/README.md (hunk 12 above)
Comment (klaudiagrz) [inline components/kyma-environment-broker/README.md]:
```suggestion
| **APP_PROVISIONING_SECRET_NAME** | Specifies the name of the Secret which holds credentials to the Provisioner's API. | None |
```

File: components/kyma-environment-broker/README.md [hunk 24]
```diff
@@ -2,21 +2,33 @@
...
+| **APP_PROVISIONING_AWS_SECRET_NAME** | Defines the name of the Secret which holds credentials to AWS. | None |
+| **APP_PROVISIONING_AZURE_SECRET_NAME** | Defines the name of the Secret which holds credentials to Azure. | None |
+| **APP_AUTH_USERNAME** | Specifies the Kyma Environment Service Broker authentication username. | None |
+| **APP_AUTH_PASSWORD** | Specifies the Kyma Environment Service Broker authentication password. | None |
+| **APP_DIRECTOR_NAMESPACE** | Specifies the Namespace in which Director is deployed. | `compass-system` |
+| **APP_DIRECTOR_URL** | Specifies the Director's URL. | `http://compass-director.compass-system.svc.cluster.local:3000/graphql` |
+| **APP_DIRECTOR_OAUTH_CREDENTIALS_SECRET_NAME** | Specifies the name of the Secret created by the Integration System. | `compass-kyma-environment-broker-credentials` |
+| **APP_DIRECTOR_SKIP_CERT_VERIFICATION** | Specifies whether TLS checks the presented certificates. | `false` |
+| **APP_DATABASE_USER** | Defines the database username. | `postgres` |
+| **APP_DATABASE_PASSWORD** | Defines the database user password | `password` |
```
Comment (kazydek) [inline components/kyma-environment-broker/README.md]:
```suggestion
| **APP_DATABASE_PASSWORD** | Defines the database user password. | `password` |
```

File: components/kyma-environment-broker/README.md [hunk 25]
```diff
@@ -2,21 +2,33 @@
...
+| **APP_DATABASE_HOST** | Defines the database host | `localhost` |
```
Comment (kazydek) [inline components/kyma-environment-broker/README.md]:
```suggestion
| **APP_DATABASE_HOST** | Defines the database host. | `localhost` |
```

File: components/kyma-environment-broker/README.md [hunk 26]
```diff
@@ -2,21 +2,33 @@
...
+| **APP_DATABASE_PORT** | Defines the database port | `5432` |
```
Comment (kazydek) [inline components/kyma-environment-broker/README.md]:
```suggestion
| **APP_DATABASE_PORT** | Defines the database port. | `5432` |
```

File: components/kyma-environment-broker/README.md [hunk 27]
```diff
@@ -2,21 +2,33 @@
...
+| **APP_DATABASE_NAME** | Defines the database name | `broker` |
```
Comment (kazydek) [inline components/kyma-environment-broker/README.md]:
```suggestion
| **APP_DATABASE_NAME** | Defines the database name. | `broker` |
```

**Pull request merged**